### PR TITLE
[Site Intent Question] Display list of default verticals on the Site Intent Question screen

### LINKED ISSUE #18107: [Site Intent Question] Display list of default verticals on the Site Intent Question screen
### Description

Subpart of:
- https://github.com/wordpress-mobile/WordPress-iOS/issues/18035

A list of clickable verticals with icons is displayed on the [Site Intent Question screen](https://github.com/wordpress-mobile/WordPress-iOS/issues/18106).

### Acceptance Criteria

- [ ] Default Verticals are bundled in the app (pbArwn-3Qf-p2#default-verticals)
  - [ ] The name of each vertical is localized
- [ ] List of verticals is displayed
  - [ ] A list item is displayed as such:
    - [ ] Emoji icon in a gray circle
    - [ ] Name of the item with bold font weight
  - [ ] List item tap → Navigates to "Choose a Design" screen

#### Other Considerations:

- [ ] Unit Tests are added where applicable
- [ ] List is scrollable
- [ ] Events are tracked (Vertical selected)

### Expected Flow

Flow2

### Expected Design

iOS_Topic_Question2

### PR DESCRIPTION
**Pull request by twstokes**
Closes #18107 

### Description

Subpart of:
- https://github.com/wordpress-mobile/WordPress-iOS/issues/18035

A list of clickable verticals with icons is displayed on the [Site Intent Question screen](https://github.com/wordpress-mobile/WordPress-iOS/issues/18106).

### Acceptance Criteria

- [x] Default Verticals are bundled in the app (pbArwn-3Qf-p2#default-verticals)
  - [x] The name of each vertical is localized
- [x] List of verticals is displayed
  - [x] A list item is displayed as such:
    - [x] Emoji icon in a gray circle
    - [x] Name of the item with bold font weight
  - [x] List item tap → Navigates to "Choose a Design" screen
- [ ] The UI scales with dynamic type (font size) changes
- [x] The UI supports Dark Mode 

Default list of verticals

#### Other Considerations:

- [x] Unit Tests are added where applicable
- [x] List is scrollable
- [x] Events are tracked (Vertical selected) ⚠️ This may create noise as multiple verticals could be selected in one session. We may consider sending the selection even at the very end of the site creation in a later PR. 

## Regression Notes
1. Potential unintended areas of impact
None currently as this is behind a feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private static let defaultVerticalSlugs = [
+        "food",
+        "news",
+        "lifestyle",
+        "personal",
+        "photography",
+        "travel"
+    ]
+
+    static let verticals: Set<SiteIntentVertical> = [
```
Comment (twstokes) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: These were a lot more readable when I added tabs to align them, but the linter got angry.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/IntentCell.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        accessibilityTraits = .button
+        accessibilityHint = NSLocalizedString("Selects this topic as the intent for your site.",
+                                              comment: "Accessibility hint for a topic in the Site Creation intents view.")
+    }
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        accessoryType = .disclosureIndicator
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/IntentCell.swift]: nit: since we're using a `xib`, maybe we can configure this in there?

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        .init("lifestyle", NSLocalizedString("Lifestyle", comment: "Lifestyle site intent topic"), "☕"),
+        .init("local_services", NSLocalizedString("Local Services", comment: "Local Services site intent topic"), "📍"),
+        .init("music", NSLocalizedString("Music", comment: "Music site intent topic"), "🎵"),
+        .init("news", NSLocalizedString("News", comment: "News site intent topic"), "🗞️"),
+        .init("parenting", NSLocalizedString("Parenting", comment: "Parenting site intent topic"), "👶"),
+        .init("people", NSLocalizedString("People", comment: "People site intent topic"), "🧑‍🤝‍🧑"),
+        .init("personal", NSLocalizedString("Personal", comment: "Personal site intent topic"), "✍️"),
+        .init("photography", NSLocalizedString("Photography", comment: "Photography site intent topic"), "📷"),
+        .init("politics", NSLocalizedString("Politics", comment: "Politics site intent topic"), "🗳️"),
+        .init("real_state", NSLocalizedString("Real Estate", comment: "Real Estate site intent topic"), "🏠"),
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: typo in the slug: `real_state` -> `real_estate`

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import Foundation
+
+struct SiteIntentData {
+    // Ordering of defaults affects ordering in the UI.
+    private static let defaultVerticalSlugs = [
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: Optional: if we want to avoid repeating the slugs (strings) of the default verticals, we could add a property in `SiteIntentVertical` that identifies a default vertical. This would increase the complexity (by either transforming `verticals` into an ordered array, or sorting the defaults afterwards), but given the small amount of entries, it should not be an issue.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift (hunk 1 above)
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: Totally fine for now, but, in a future PR, we could think of extracting the hardcoded strings into a (localized) .plist so that we pull info from that file transparently.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    static let defaultVerticals: [SiteIntentVertical] = {
+        return defaultVerticalSlugs.compactMap { slug in
+            verticals.first(where: { $0.slug == slug })
+        }
+    }()
+}
+
+fileprivate extension SiteIntentVertical {
+    init(_ slug: String, _ localizedTitle: String, _ emoji: String) {
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: any reason this initializer should only be visible here?

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift [hunk 6]
```diff
@@ -4,12 +4,16 @@ class SiteIntentViewController: CollapsableHeaderViewController {
...
     private let table: UITableView
 
-    private var selectedVertical: SiteVertical? {
+    private var selectedVertical: SiteIntentVertical? {
         didSet {
             itemSelectionChanged(selectedVertical != nil)
         }
     }
 
+    private var data: [SiteIntentVertical] {
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: maybe we could rename this property to be more specific? `data` sounds kinda generic, we could use `availableVerticals` or something like that, what do you think?

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift [hunk 7]
```diff
@@ -65,15 +70,23 @@ class SiteIntentViewController: CollapsableHeaderViewController {
         navigationItem.leftBarButtonItem = UIBarButtonItem(title: NSLocalizedString("Cancel", comment: "Cancel site creation"), style: .done, target: self, action: #selector(closeButtonTapped))
     }
 
+    private func setupTable() {
+        table.dataSource = self
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: we should probably do this at initialization time (after `super.init()`), what do you think?

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift [hunk 8]
```diff
@@ -4,12 +4,16 @@ class SiteIntentViewController: CollapsableHeaderViewController {
     private let selection: SiteIntentStep.SiteIntentSelection
     private let table: UITableView
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: nit: I know this was not part of this PR, but can we change the property name from `table` to something that more clearly recalls that it's a `UIKit` object? Maybe just `tableView` or `siteIntentsTableview`?

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift [hunk 9]
```diff
@@ -88,3 +101,41 @@ class SiteIntentViewController: CollapsableHeaderViewController {
...
+    }
+}
+
+// MARK: UITableViewDelegate
+
+extension SiteIntentViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        // TODO - either use or remove after implementing search
+        // searchTextField.resignFirstResponder()
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: Since this will be addressed in a future PR, maybe we can just remove the commented code for now? Just in case, for any reason, it gets committed and then we don't handle it. Also, maybe add a specific keyword to `TODOs` (e.g.`TODO: - SITECREATION - [description]` makes them more searchable afterwards, what do you think?

**Review by Gio2018 — CHANGES_REQUESTED** [review]
Hey @twstokes , I tested on iPhone and iPad and it worked well! I left some code comments for your consideration.
One additional consideration:
- The UI scales mostly fine, but, as discussed on Slack, the nav bar button items have a weird scaling behavior, will open a specific issue for that.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift (hunk 1 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: That sounds like a much cleaner approach @Gio2018!

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/IntentCell.swift (hunk 2 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/IntentCell.swift]: Agreed! Done.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift (hunk 3 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: Thanks! 😬 I've fixed this in the source document as well.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift (hunk 5 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: I made it private to this file because I only saw us using it here to import the data. Typically I'd want to provide the member names. What do you think?

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift (hunk 6 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: Sure. I was keeping some similarities between this site creation step and [existing ones](https://github.com/wordpress-mobile/WordPress-iOS/blob/feature/site-intent-default-verticals/WordPress/Classes/ViewRelated/Site%20Creation/WebAddress/WebAddressWizardContent.swift#L41) but I doubt we'll get a lot of value out of that.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift (hunk 4 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: I don't have a strong opinion on this since the dataset is so small, so I've added the `isDefault` property. The `verticals` set is now an array, and I've moved the defaults up to the top and ordered them since they aren't alphabetical.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift (hunk 7 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: Good call, fixed. [Same issue here](https://github.com/wordpress-mobile/WordPress-iOS/blob/7195beae464dae36b94d3951f807907e0f039b9e/WordPress/Classes/ViewRelated/Site%20Creation/WebAddress/WebAddressWizardContent.swift#L329).

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift (hunk 8 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: Renamed to `tableView`. 👍

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift (hunk 9 above)
Comment (twstokes) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift]: I've removed TODOs and stubs.

File: WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift (hunk 5 above)
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentData.swift]: Makes sense! Thanks for clarifying

**Review by Gio2018 — APPROVED** [review]
Looks great @twstokes !
:shipit: !
